src/manifold/src/edge_op.cpp

```diff
@@ -380,6 +427,10 @@ void Manifold::Impl::RemoveIfFolded(int edge) {
   }
 }
 
+// Collapses the given edge by removing startVert. May split the mesh
+// topologically if the collapse would have resulted in a 4-manifold edge. Do
+// not collapse an edge if startVert is pinched - the vert will be marked NaN,
+// but other edges may still be pointing to it.
```

Comment:
This was the primary cause of the seg fault, which I fixed in several ways. This problem has existed for quite some time, but was illuminated by unusual triangulations.